src/disclosure-manager/DisclosureManagerContext.jsx

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import { DisclosureManagerContext } from 'terra-disclosure-manager';
+
+const contextShape = PropTypes.shape({
+  /**
+   * A function that will disclose content within the determined DisclosureManager.
+   * Ex:
+   *
+   *  disclose({
+   *    preferredType: 'modal',
+   *    size: 'large',
+   *    content: {
+   *      key: 'example-modal',
+   *      component: <ExampleModal />
+   *    },
+   *  })
+   */
+  disclose: PropTypes.func.isRequired,
+
+  /**
+   * The following properties are only available to components that have been
+   * disclosed by a Disclosure Manager.
+   */
+
+  /**
+   * A function that will dismiss the active disclosed component and remove it
+   * from the disclosure stack.
+   */
+  dismiss: PropTypes.func,
+  /**
+   * A function that will dismiss all components within the disclosure stack.
+   */
+  closeDisclosure: PropTypes.func,
+  /**
+   * A function that will dismiss the active disclosed component and remove it
+   * from the disclosure stack. Similar to `dismiss`, though `goBack` is only
+   * provided to nested disclosure components (i.e. when there is a component to
+   * go back to).
+   */
+  goBack: PropTypes.func,
+  /**
+   * A function that will maximize the size of the disclosure. This function will
+   * not be provided if the disclosure is already maximized.
+   */
+  maximize: PropTypes.func,
+  /**
+   * A function that will minimize the size of the disclosure. This function will
+   * not be provided if the disclosure is already minimized.
+   */
+  minimize: PropTypes.func,
+  /**
+   * A function that will allow a component to register a function with the
+   * DisclosureManager that will be called before the component is dismissed for
+   * any reason. This function is expected to return true if the dismissal should
+   * be allowed or false otherwise.
+   */
+  registerDismissCheck: PropTypes.func,
```

Comment:
If a consumer provided their own `registerDismissCheck`, will it override the pending data checking?

Reply:
It definitely would, for that disclosed component at least. The expectation is that components that override the default value would take responsibility over the pending actions.

I should probably call that out more explicitly in the documentation.

Reply:
Updated here: https://github.com/cerner/terra-application/pull/2/files#diff-feeb949f26ffe443e6fd52d1128878edR125